File: src/main/java/pl/training/workshop/m6/s10_implicittree/step3/BarMenu.java
package pl.training.workshop.m6.s10_implicittree.step3;

import java.util.List;

import pl.training.workshop.shared.Money;

/**
 * Krok 3: render też na Composite. BarMenu tylko mapuje stary format na drzewo - format
 * trwały (zagnieżdżone listy) zmienimy osobnym krokiem, jeśli w ogóle.
 */
public final class BarMenu {
    public Money price(List<?> combo) {
        return MenuMapper.fromNested(combo).price();
    }

    public String render(List<?> combo) {
        StringBuilder text = new StringBuilder();
        MenuMapper.fromNested(combo).render(0, text);
        return text.toString();
    }
}

File: csharp/src/Training.Workshop/M6/S10ImplicitTree/Step3/BarMenu.cs
using System.Text;
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S10ImplicitTree.Step3;

/// <summary>
/// Krok 3: Render też na Composite. BarMenu tylko mapuje stary format na drzewo - format
/// trwały (zagnieżdżone listy) zmienimy osobnym krokiem, jeśli w ogóle.
/// </summary>
public sealed class BarMenu
{
    public Money Price(IReadOnlyList<object> combo)
    {
        return MenuMapper.FromNested(combo).Price;
    }

    public string Render(IReadOnlyList<object> combo)
    {
        var text = new StringBuilder();
        MenuMapper.FromNested(combo).Render(0, text);
        return text.ToString();
    }
}
